src/GraphQL.Client.Serializer.SystemTextJson/PolymorphicTypenameConverter.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace GraphQL.Client.Serializer.SystemTextJson
+{
+    public abstract class PolymorphicTypenameConverter<T> : JsonConverter<T>
+    {
+        protected abstract Type Descriminator(string typename);
+
+        public override bool CanConvert(Type typeToConvert) =>
+            typeof(T).IsAssignableFrom(typeToConvert);
+
+        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var clone = reader; // cause its a struct

```

Comment:
?

Reply:
Ok. Better to put a comment into sources.

Reply:
By itself, this class without descendants carries little benefit. You can make at least one descendant, which discovers all the classes in the declaring assembly.